src/Device/Driver/ATR833.cpp

```diff
@@ -25,15 +25,36 @@ Copyright_License {
 #include "Device/Driver.hpp"
 #include "Device/Port/Port.hpp"
 #include "RadioFrequency.hpp"
+#include "NMEA/Info.hpp"
+#include "Operation/Operation.hpp"
+#include "time/TimeoutClock.hpp"
+#include "thread/Mutex.hxx"
+#include "util/StaticFifoBuffer.hxx"
+
+#include <cstdint>
+
+#include <stdio.h>
 
 static constexpr uint8_t STX = 0x02;
 static constexpr uint8_t SYNC = 'r';
 
 class ATR833Device final : public AbstractDevice {
+  static constexpr uint8_t ACK = 0x06;
+  static constexpr uint8_t NAK = 0x15;
+  static constexpr uint8_t ALIVE = 0x10;
+  static constexpr uint8_t EXCHANGE = 0x11;
   static constexpr uint8_t SETSTANDBY = 0x12;
   static constexpr uint8_t SETACTIVE = 0x13;
+  static constexpr uint8_t ALLDATA = 0x42;
+  static constexpr uint8_t REQUESTDATA = 0x82;
 
   Port &port;
+  // Expected length of the message just receiving.
+  size_t expected_msg_length{};
+  // Buffer which receives the messages send from the radio.
+  StaticFifoBuffer<uint8_t, 256u> rx_buf;
```

Comment:
Why a FIFO buffer? A FIFO buffer can wrap around the end, but your code looks like it can't handle that.